lib/ansible/modules/cloud/amazon/iam_cert_facts.py

```diff
@@ -0,0 +1,161 @@
+'''
+
+RETURN = '''
+is_truncated:
+    description: A flag that indicates whether there are more items to return.
+    returned: always
+    type: bool
+    sample: false
+
+response_metadata:
+    description: Response metadata.
+    returned: always
+    type: dict
+    sample: {"http_status_code": 200, "request_id": "9f6d-671298e72d8b"}
+
+server_certificate_metadata_list:
+    description: Metadata list.
+    returned: always
+    type: dict
+    sample: http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_server_certificates
+'''
+
+try:
+    import boto3
+    import botocore
+
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (
+    boto3_conn, ec2_argument_spec, get_aws_connection_info,
+    camel_dict_to_snake_dict)
+
+
+def get_server_certificate(client, module, server_certificate_name):
+    """ Retrieves information about the specified server certificate stored in IAM.
+    """
+    try:
+        response = client.get_server_certificate(
+            ServerCertificateName=server_certificate_name)
+    except ClientError as e:
+        client_error = camel_dict_to_snake_dict(e.response)
+        if client_error['error']['code'] == 'NoSuchEntity':
+            return client_error['error']
+        else:
+            raise
+    except Exception as e:
+        module.fail_json(msg="Unable to get server certificate :: %s" % str(e))
+    return response
+
+
+def list_server_certificates(client, module, path_prefix):
+    """ Lists the server certificates stored in IAM that have the specified path prefix.
+        If none exist, the action returns an empty list.
+    """
+    try:
+        if path_prefix:
+            response = client.list_server_certificates(PathPrefix=path_prefix)
+        else:
+            response = client.list_server_certificates()
+    except Exception as e:
+        module.fail_json(msg="Unable to get server certificates :: %s" % str(e))
+    return response
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        certificate_name=dict(type='str', default=None),
+        path_prefix=dict(type='str', default=None),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['certificate_name', 'path_prefix'],
+        ],
+        supports_check_mode=False,
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg="boto3 is required")
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        iam_cert = boto3_conn(module, conn_type='client', resource='iam',
+                              region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except Exception as e:
+        module.fail_json(msg="Can't authorize connection - %s " % str(e))
```

Comment:
If you're going to catch Exception, please include a traceback by adding `, exception=traceback.format_exc()` to the fail_json so it can be debugged. You'll also have to `import traceback`.